### PR TITLE
feat: Add team member role management

### PR DESCRIPTION
**Pull request by gentamura**
## Summary
Add team member role management functionality with proper validation and authorization checks.

## Related Issue
N/A

## Changes
- Add team member role editing functionality
  - Implement inline role editing UI
  - Add role update server action
  - Add type safety with proper validation
- Add authorization checks
  - Implement admin role validation
  - Add validation to prevent removing the last admin
  - Add user ID type guard
- Improve code structure
  - Replace direct team queries with fetchCurrentTeam service
  - Add count function from drizzle-orm for admin counting
  - Fix type errors in role management

## Testing
- Tested role update functionality
  - Admin role validation
  - Last admin protection
  - Type guard validations
- Verified proper error handling
  - Invalid role changes
  - User ID validation
  - Authorization checks

## Other Information
The PR ensures that:
- Team always maintains at least one admin
- Only valid user IDs are processed
- Proper type safety is maintained throughout
- Server-side validation complements UI restrictions

### CODE REVIEW

**Comment by gentamura** [comment]
@satococoa review, please 🙏

File: app/(main)/settings/team/actions.ts [hunk 1]
```diff
@@ -180,6 +185,117 @@ export async function addTeamMember(formData: FormData) {
...
+			.from(users)
+			.where(eq(users.id, userId))
+			.limit(1);
+
+		if (user.length === 0) {
+			throw new Error("User not found");
+		}
+
+		// 4. If changing from admin to member, check if this is the last admin
+		if (role === "member") {
```
Comment (satococoa) [inline app/(main)/settings/team/actions.ts]:
```suggestion
		if (role === "member" && ! isUpdatingSelf) {
```

This validation is only necessary when updating one's own role.
When changing another user's role to "member", there will always be at least one admin (the current user) remaining.

And SQL may be simpler. Only check if there is at least one admin user other than the current user in the team.

**Review by satococoa — COMMENTED** [review]
👀 Left some comments here

File: app/(main)/settings/team/actions.ts (hunk 1 above)
Comment (gentamura) [reply app/(main)/settings/team/actions.ts]: @satococoa 
Thanks for the review 👍 
Based on the feedback, I've simplified the code as shown in commit 279763c93c7c6f672a5cf5f8a149ab2ea6bb8602 :pray:

**Review by satococoa — APPROVED** [review]
Thank you! 👍

**Comment by gentamura** [comment]
@shige @toyamarinyon review, please :pray:

**Review by toyamarinyon — APPROVED** [review]

File: app/(main)/settings/team/team-members.tsx [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
+import { getCurrentUserRole, getTeamMembers } from "./actions";
 import { TeamMembersForm } from "./team-members-form";
 import { TeamMembersList } from "./team-members-list";
 
 export async function TeamMembers() {
 	const team = await fetchCurrentTeam();
 	const { data: teamRole } = await getCurrentUserRole();
 
+	const result = await getTeamMembers();
+	const currentUserRoleResult = await getCurrentUserRole();
```
Comment (shige) [inline app/(main)/settings/team/team-members.tsx]: nits:

It seems calling the same method in L9, but is it possible to consolidate it here without calling it again?

**Review by shige — APPROVED** [review]
LGTM 👍 

---

NOTE:

The overall number of calls to `supabase.auth.getUser()` per request is increasing, so I will create a new issue for re-architecting it separately.

**Comment by shige** [comment]
> NOTE:
> 
> The overall number of calls to `supabase.auth.getUser()` per request is increasing, so I will create a new issue for re-architecting it separately.

- 🔗 https://github.com/giselles-ai/giselle/issues/208

File: app/(main)/settings/team/team-members.tsx (hunk 2 above)
Comment (gentamura) [reply app/(main)/settings/team/team-members.tsx]: Thanks! fixed 09729f20410e

**Comment by gentamura** [comment]
Thanks for reviewers 🚀

**Pull request merged**